frontend/src/component/common/Card/Card.tsx

```diff
@@ -1,6 +1,71 @@
-import { styled, Card as MUICard, Box } from '@mui/material';
+import {
+    styled,
+    Card as MUICard,
+    Box,
+    type CardProps,
+    type Theme,
+    type CSSObject,
+} from '@mui/material';
 
-const StyledCard = styled(MUICard)(({ theme }) => ({
+type CardVariant = 'primary' | 'secondary';
+type VariantProps = {
+    [key in CardVariant]: {
+        card(theme: Theme): CSSObject;
+        header(theme: Theme): CSSObject;
+        body(theme: Theme): CSSObject;
+        footer(theme: Theme): CSSObject;
+    };
+};
+const variants: VariantProps = {
+    primary: {
+        card: (theme: Theme): CSSObject => ({
+            backgroundColor: theme.palette.background.default,
+            background: theme.palette.secondary.light,
+            '&:hover': {
+                backgroundColor: theme.palette.neutral.light,
+            },
+        }),
+        header: (theme: Theme): CSSObject => ({
+            fontWeight: theme.typography.fontWeightRegular,
+        }),
+        body: (theme: Theme): CSSObject => ({
+            backgroundColor: theme.palette.background.default,
+            color: theme.palette.text.secondary,
+            '&:hover': {
+                backgroundColor: theme.palette.neutral.light,
+            },
+            fontSize: theme.fontSizes.smallBody,
+        }),
+        footer: (theme: Theme): CSSObject => ({
+            borderTop: `1px solid ${theme.palette.divider}`,
+            background: theme.palette.background.elevation1,
+            boxShadow: theme.boxShadows.accordionFooter,
+            color: theme.palette.text.secondary,
+        }),
+    },
+    secondary: {
+        card: (theme: Theme): CSSObject => ({
+            backgroundColor: theme.palette.secondary.light,
+            borderColor: theme.palette.secondary.border,
+            color: theme.palette.text.primary,
+        }),
+        header: (theme: Theme): CSSObject => ({
+            fontWeight: theme.typography.fontWeightBold,
+        }),
+        body: (theme: Theme): CSSObject => ({
+            backgroundColor: theme.palette.secondary.light,
+            color: theme.palette.text.primary,
+        }),
+        footer: (theme: Theme): CSSObject => ({
+            background: theme.palette.secondary.light,
+            fontSize: theme.fontSizes.bodySize,
+        }),
+    },
+};
```

Comment:
I think I would rely more on setting the defaults (primary) in the styled components and only specifying the variant overrides here.

Reply:
It has some defaults, but I wanted to be intentional about the things each one overrides, that's why there's no default to `fontSize` or `color` (note: there's always a default). But if it's something different than the base CSS I want it to be explicit, not a default of the component.